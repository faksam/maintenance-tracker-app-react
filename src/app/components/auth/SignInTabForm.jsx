import React from 'react';

import PropTypes from 'prop-types';

/**
 * @class SignInTabForm
 * @extends {Component}
 *
 * @description Displays the list of replies to a comment
 *
 */
const SignInTabForm = ({
  handleSignIn, accountEmail, accountPassword, handleChange,
  errorDivClass, errors
}) => (
  <div>
    <div className="container">
      <h2 className="float-left">Sign In</h2>
      <form id="loginForm" className="center" method="POST">
        <div className="row">
          <div className="row">
            <div className="input col col-1">
              <i className="material-icons">email</i>
              &nbsp;
              <input
                id="accountEmail"
                name="accountEmail"
                type="email"
                placeholder="Email"
                value={accountEmail}
                onChange={handleChange}
                required
              />
              <div id="errorDiv" className={errorDivClass}>
                {
                  errors.email
                    ? (<p key="email" className="danger">{errors.email}</p>)
                    : ''
                }
                <p id="signupErrorMessage" className="hide-div" />
              </div>
            </div>
          </div>
          <div className="row">
            <div className="input col col-1">
              <i className="material-icons">account_circle</i>
              &nbsp;
              <input
                id="accountPassword"
                name="accountPassword"
                type="password"
                placeholder="Password"
                value={accountPassword}
                onChange={handleChange}
                required
              />
              <div id="errorDiv" className={errorDivClass}>
                {
                  errors.password
                    ? (<p key="password" className="danger">{errors.password}</p>)
                    : ''
                }
                <p id="signupErrorMessage" className="hide-div" />
              </div>
            </div>
          </div>
          <div className="row">
            <div className="input col col-1">
              <button className="btn-signin"
                onClick={handleSignIn}
                type="button"
              >
                Sign In
              </button>
            </div>
          </div>
        </div>
      </form>
      <br />
    </div>
  </div>
);

SignInTabForm.propTypes = {
  errorDivClass: PropTypes.string,
  errors: PropTypes.instanceOf(Object),
  handleSignIn: PropTypes.func.isRequired,
  accountEmail: PropTypes.string.isRequired,
  accountPassword: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default SignInTabForm;
