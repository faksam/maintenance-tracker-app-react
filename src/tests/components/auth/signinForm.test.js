import React from 'react';
import { shallow } from 'enzyme';
import SignInTabForm from '../../../app/components/auth/SignInTabForm';

const props = {
  accountEmail: 'fakunlesamuel@gmail.com',
  accountPassword: 'Lifeisarace',
  handleSignIn: jest.fn(),
  handleChange: jest.fn(),
};
describe('AuthForms Component', () => {
  it('should render without throwing an error', () => {
    const wrapper = shallow(<SignInTabForm {...props} />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
