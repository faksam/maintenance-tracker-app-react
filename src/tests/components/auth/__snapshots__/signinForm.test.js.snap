// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForms Component should render without throwing an error 1`] = `
<div>
  <div
    className="container"
  >
    <h2
      className="float-left"
    >
      Sign In
    </h2>
    <form
      className="center"
      id="loginForm"
      method="POST"
    >
      <div
        className="row"
      >
        <div
          className="row"
        >
          <div
            className="input col col-1"
          >
            <i
              className="material-icons"
            >
              email
            </i>
             
            <input
              id="accountEmail"
              name="accountEmail"
              onChange={[MockFunction]}
              placeholder="Email"
              required={true}
              type="email"
              value=""
            />
            <div
              className="display-none"
              id="errorDiv"
            >
              <p
                className="hide-div"
                id="signupErrorMessage"
              />
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input col col-1"
          >
            <i
              className="material-icons"
            >
              account_circle
            </i>
             
            <input
              id="accountPassword"
              name="accountPassword"
              onChange={[MockFunction]}
              placeholder="Password"
              required={true}
              type="password"
              value=""
            />
            <div
              className="display-none"
              id="errorDiv"
            >
              <p
                className="hide-div"
                id="signupErrorMessage"
              />
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input col col-1"
          >
            <button
              className="btn-signin"
              onClick={[MockFunction]}
              type="button"
            >
              Sign In
            </button>
          </div>
        </div>
      </div>
    </form>
    <br />
  </div>
</div>
`;
